pxr/base/tf/stringUtils.h

```diff
@@ -683,18 +727,113 @@ std::string TfStringCatPaths( const std::string &prefix,
 inline bool
 TfIsValidIdentifier(std::string const &identifier)
 {
-    char const *p = identifier.c_str();
-    auto letter = [](unsigned c) { return ((c-'A') < 26) || ((c-'a') < 26); };
-    auto number = [](unsigned c) { return (c-'0') < 10; };
-    auto under = [](unsigned c) { return c == '_'; };
-    unsigned x = *p;
-    if (!x || number(x)) {
-        return false;
+    if (UseUTF8Identifiers())
```

Comment:
I think that the general tactic we'll need to go with in order to avoid performance fallout is to optimistically take the ASCII path, and only if we see a byte with the high-order bit set do we hop off onto a slower UTF-8 path.

Reply:
Implementation now uses current sorting algorithm with the addition of UTF-8 byte sorting when there is a mismatch.